src/module/auth/auth.controller.ts

```diff
@@ -27,4 +42,31 @@ export class AuthController {
       tokens: result.tokens,
     };
   }
+
+  @Get('/github')
+  @UseGuards(AuthGuard(AuthProviderEnum.GITHUB))
+  githubLogin() {}
+
+  @Get('/github/callback')
+  @Redirect(`http://localhost:5173`)
+  @UseGuards(AuthGuard(AuthProviderEnum.GITHUB))
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  async githubCallback(@Req() req, @Res() res) {
+    const user = await this.userService.upsertUser({
+      email: req.user.profile.email,
+    });
+    const { accessToken, refreshToken} = await this.magicLinkService.generateTokens(
+      user[0].email,
+      user[0]._id,
+    );
+    return {
```

Comment:
I don't think this will return json response the way you're thinking it will.
if we have called `@Res() res` in parameter we can't directly return object
it won't be converted to json by nest
we have to explicitly use `res.json({})` to return json in such cases